packages/core/src/saml-applications/routes/index.ts

```diff
@@ -137,4 +145,103 @@ export default function samlApplicationRoutes<T extends ManagementApiRouter>(
       return next();
     }
   );
+
+  router.post(
+    '/saml-applications/:id/secrets',
+    koaGuard({
+      params: z.object({ id: z.string() }),
+      body: z.object({ lifeSpanInDays: z.number().optional() }),
+      response: samlApplicationSecretResponseGuard,
+      status: [201, 400, 404],
+    }),
+    async (ctx, next) => {
+      const {
+        body: { lifeSpanInDays },
+        params: { id },
+      } = ctx.guard;
+
+      ctx.status = 201;
+      ctx.body = await createSamlApplicationSecret(id, lifeSpanInDays);
+
+      return next();
+    }
+  );
+
+  router.get(
+    '/saml-applications/:id/secrets',
+    koaGuard({
+      params: z.object({ id: z.string() }),
+      response: samlApplicationSecretResponseGuard.array(),
+      status: [200, 400, 404],
+    }),
+    async (ctx, next) => {
+      const { id } = ctx.guard.params;
+
+      ctx.status = 200;
+      ctx.body = await findSamlApplicationSecretsByApplicationId(id);
+
+      return next();
+    }
+  );
+
+  router.delete(
+    '/saml-applications/:id/secrets/:secretId',
```

Comment:
nit: Need to check with Rany. If I recall properly our OIDC app secrets accept a unique secret name, and the delete API is delete secret by name.

Reply:
Only show status, expiresAt, and fingerprint. Since we've provide util method for calculating fingerprint in [PR#6833](https://github.com/logto-io/logto/pull/6833), will update these secret-related APIs response later, see LOG-10525.
![image](https://github.com/user-attachments/assets/7e53db39-5008-4125-95e5-597fae3e953f)